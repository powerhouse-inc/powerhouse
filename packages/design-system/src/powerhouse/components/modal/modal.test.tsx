import { fireEvent, render, screen } from "@testing-library/react";
import { Modal } from ".";

describe("Modal Component", () => {
  it("should match snapshot", () => {
    render(
      <Modal data-testid="modal" open>
        <div>Modal Content</div>
      </Modal>,
    );

    const modalComponent = screen.getByTestId("modal");

    expect(modalComponent).toMatchSnapshot();
  });

  it("should display modal content when open", () => {
    render(
      <Modal open>
        <div>Modal Content</div>
      </Modal>,
    );

    expect(screen.getByText("Modal Content")).toBeInTheDocument();
  });

  it("should call onClose callback when press esc", () => {
    const onClose = vi.fn();
    render(
      <Modal onOpenChange={onClose} open>
        <div>Modal Content</div>
      </Modal>,
    );

    fireEvent.keyDown(screen.getByText("Modal Content"), {
      key: "Escape",
      code: "Escape",
    });

    expect(onClose).toHaveBeenCalled();
  });
});
