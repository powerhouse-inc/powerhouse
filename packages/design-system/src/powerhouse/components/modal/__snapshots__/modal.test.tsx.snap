// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Component > should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in"
  data-state="open"
  data-testid="modal"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <span
    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    <h2
      id="radix-:r1:"
    >
      Modal
    </h2>
    <p
      id="radix-:r2:"
    >
      Modal
    </p>
  </span>
  <div>
    Modal Content
  </div>
</div>
`;
